cloud-core/src/main/java/cloud/commandframework/arguments/standard/DurationArgument.java

```diff
@@ -0,0 +1,269 @@
+
+        private Builder(final @NonNull String name) {
+            super(Duration.class, name);
+        }
+
+        /**
+         * Builder a new boolean component
+         *
+         * @return Constructed component
+         */
+        @Override
+        public @NonNull DurationArgument<C> build() {
+            return new DurationArgument<>(this.isRequired(), this.getName(), this.getDefaultValue(),
+                    this.getSuggestionsProvider(), this.getDefaultDescription()
+            );
+        }
+
+    }
+
+
+    public static final class DurationParser<C> implements ArgumentParser<C, Duration> {
+
+        @Override
+        public @NonNull ArgumentParseResult<Duration> parse(
+                final @NonNull CommandContext<C> commandContext,
+                final @NonNull Queue<String> inputQueue
+        ) {
+            final String input = inputQueue.peek();
+            if (input == null) {
+                return ArgumentParseResult.failure(new NoInputProvidedException(
+                        DurationArgument.DurationParseException.class,
+                        commandContext
+                ));
+            }
+            inputQueue.remove();
+
+            final Matcher matcher = DURATION_PATTERN.matcher(input);
+
+            Duration duration = Duration.ofNanos(0);
+
+            while (matcher.find()) {
+                String group = matcher.group();
+                String timeUnit = String.valueOf(group.charAt(group.length() - 1));
+                int timeValue = Integer.parseInt(group.substring(0, group.length() - 1));
+                switch (timeUnit) {
+                    case "d":
+                        duration = duration.plusDays(timeValue);
+                        break;
+                    case "h":
+                        duration = duration.plusHours(timeValue);
+                        break;
+                    case "m":
+                        duration = duration.plusMinutes(timeValue);
+                        break;
+                    case "s":
+                        duration = duration.plusSeconds(timeValue);
+                        break;
+                    default:
+                        return ArgumentParseResult.failure(new DurationArgument.DurationParseException(input, commandContext));
+                }
+            }
+
+            if (duration.isZero()) {
+                return ArgumentParseResult.failure(new DurationArgument.DurationParseException(input, commandContext));
+            }
+
+            return ArgumentParseResult.success(duration);
+        }
+
+        @Override
+        @SuppressWarnings("MixedMutabilityReturnType")
+        public @NonNull List<@NonNull String> suggestions(
+                final @NonNull CommandContext<C> commandContext,
+                final @NonNull String input
+        ) {
+            char[] chars = input.toLowerCase(Locale.ROOT).toCharArray();
+
+            Stream<String> nums = IntStream.range(1, 10).boxed()
```

Comment:
Can't you create this collection (I don't see why exactly you're creating a stream which you are collecting directly afterwards) once and store it? Same for the units below?

Reply:
Ah good point, forgot I could refactor those after I made a change earlier, thanks!